mountpoint-s3-fs/src/fs/handles.rs

```diff
@@ -4,27 +4,57 @@ use mountpoint_s3_client::types::ETag;
 use mountpoint_s3_client::ObjectClient;
 use tracing::{debug, error};
 
+use super::{Error, InodeNo, OpenFlags, S3Filesystem, ToErrno};
+use crate::metablock::LookUp;
+use crate::metablock::S3Location;
 use crate::object::ObjectId;
 use crate::prefetch::PrefetchGetObject;
-use crate::superblock::path::ValidKey;
-use crate::superblock::{LookedUp, ReaddirHandle};
+use crate::superblock::LookedUpInode;
+use crate::superblock::ReaddirHandle;
 use crate::sync::atomic::{AtomicI64, Ordering};
 use crate::sync::AsyncMutex;
 use crate::upload::{AppendUploadRequest, UploadRequest};
+use std::ffi::OsString;
 
-use super::{DirectoryEntry, Error, InodeNo, OpenFlags, S3Filesystem, ToErrno};
+#[derive(Debug, Clone)]
+pub struct DirectoryEntryReaddir {
```

Comment:
This and `DirHandle` could move into the `superblock` module. But let's leave it as the last commit.